### PR TITLE
APP-1502: Add flag for payments screen in profile tab

### PR DESCRIPTION
**Pull request by oscarnylander**
Bonus: Add a cheeky test for the charity-flag as well

### CODE REVIEW

**Review by StylianosGakis — APPROVED** [review]
Ah I am so glad that this was so simple to add on top after the changes we've done before. Love it!

File: app/src/test/java/com/hedvig/app/feature/profile/ui/tab/ProfileQueryDataToProfileUiStateMapperTest.kt [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    fun `when charity-feature is activated, should show charity-data`() = runTest {
+        val featureManager = mockk<FeatureManager>(relaxed = true)
+        coEvery { featureManager.isFeatureEnabled(Feature.SHOW_CHARITY) } returns true
+        val mapper = sut(
+            featureManager = featureManager
+        )
+
+        val result = mapper.map(PROFILE_DATA)
+
+        assertThat(result.charityState).isNotInstanceOf(CharityState.DontShow::class)
```
[the remark this answers is not in the document]
Comment (hedvigoscar) [reply app/src/test/java/com/hedvig/app/feature/profile/ui/tab/ProfileQueryDataToProfileUiStateMapperTest.kt:76]: image

File: app/src/test/java/com/hedvig/app/feature/profile/ui/tab/ProfileQueryDataToProfileUiStateMapperTest.kt [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+import com.hedvig.app.testdata.feature.profile.PROFILE_DATA
+import com.hedvig.app.util.LocaleManager
+import com.hedvig.app.util.featureflags.FeatureManager
+import com.hedvig.app.util.featureflags.flags.Feature
+import io.mockk.coEvery
+import io.mockk.mockk
+import kotlinx.coroutines.test.runTest
+import org.junit.Test
+
+class ProfileQueryDataToProfileUiStateMapperTest {
```
[the remark this answers is not in the document]
Comment (hedvigoscar) [reply app/src/test/java/com/hedvig/app/feature/profile/ui/tab/ProfileQueryDataToProfileUiStateMapperTest.kt:16]: 😉

**Comment by hedvigoscar** [comment]
Good architecture makes introducing changes easy. 🚀

File: app/src/test/java/com/hedvig/app/feature/profile/ui/tab/ProfileQueryDataToProfileUiStateMapperTest.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (StylianosGakis) [reply app/src/test/java/com/hedvig/app/feature/profile/ui/tab/ProfileQueryDataToProfileUiStateMapperTest.kt:76]: Oh I completely missed that 😅 of course 🧠🧠🧠

**Pull request merged**